src/Domain/WeightedElementDistribution.hpp

```diff
@@ -0,0 +1,147 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#pragma once
+
+#include <array>
+#include <cstddef>
+#include <unordered_set>
+#include <utility>
+#include <vector>
+
+template <size_t Dim>
+class Block;
+
+template <size_t Dim>
+class ElementId;
+
+namespace Spectral {
+enum class Quadrature;
+}  // namespace Spectral
+
+namespace domain {
+
+/*!
+ * \brief Distribution strategy for assigning elements to CPUs using a
+ * Morton ('Z-order') space-filling curve to determine placement within each
+ * block, where elements are assigned weighted costs that are distributed
+ * across CPUs
+ *
+ * \details The element distribution attempts to assign a balanced total
+ * computational cost to each processor that is allowed to have elements. The
+ * cost assigned to a processor is defined as the sum of the costs of the
+ * `Element`s assigned to a processor, and the cost of an `Element` is defined
+ * as `(number of grid points) / sqrt(minimum grid spacing in Frame::Grid)`.
```

Comment:
Where does `sqrt(min grid spacing)` come from? Isn't the time step size given by the minimum grid spacing? Could you add a (brief!) explanation into the dox as to where this comes from?

Reply:
Yep! By "where it comes from" are you wanting me to comment on why minimum grid spacing is being used or why the square root is there? Or both?